Reader: Fixes an issue with reader comments being incorrectly rendered.

aerych opened the pull request:
This PR is part of a bug fix for WPiOS `release/17.8`. 
More in p1626841091202200-slack-hummingbird

WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/419

In https://github.com/wordpress-mobile/WordPressKit-iOS/pull/413 we made a change to consume `raw_content` rather than `content` field from the Comments endpoints.  This [resolved](https://github.com/wordpress-mobile/WordPress-iOS/pull/16768) an [issue](https://github.com/wordpress-mobile/WordPress-iOS/issues/16685) with Authors and Contributors not being able to view Comment content.  However, we missed that it created a new issue with comments in the Reader being incorrectly rendered in many cases as the `raw_comment` field does not include html mark up for non-P2 WordPress sites.  P2s that use the block editor for comments were rendered correctly.  We were unaware of this difference between P2 and non-P2 comments.  The reader's caching behavior likely contributed to missing the regression. 

Example of the badness. Note the cropped lines of text:
![Simulator Screen Shot - iPhone 11 - 2021-07-20 at 23 12 22](https://user-images.githubusercontent.com/1435271/126558810-7ee0e22d-6513-4a28-879a-1cba628e4407.png)


This PR adopts changes in WordPressKit to consume both `content` and `raw_content` fields returned by the comments endpoint.  Blog comments that are wrangled by the My Site and Notifications feature continue to use the `raw_content` value.  Reader comments are assigned the `content` value to preserve the expected HTML mark up and restore the correct rendering behavior.   The changes are a temporary measure that we'd like to improve in a subsequent update.

### Testing Details
Switch to this branch and test the following: 
- Confirm that reader comments are correctly rendered.  
- Confirm that comments in My Sites are correctly rendered.
- Confirm that authors and contributors can still view comments under My Sites.
- Confirm that comments under Notifications are correctly rendered.
- Confirm that replying to a reader comment works as expected and there are no layout issues. 
- Confirm that editing a my sites or notifications comment works as expected.

## Regression Notes
1. Potential unintended areas of impact
Non-reader comments.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Smoke tested Notifications and My Site comments.

3. What automated tests I added (or what prevented me from doing so)
Requires a visual evaluation.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

aerych committed:
Reader: Fixes an issue with reader comments being incorrectly rendered.

ScoutHarris commented on WordPress/Classes/Services/CommentService.m:
```diff
@@ -1132,6 +1132,39 @@ - (Comment *)findCommentWithID:(NSNumber *)commentID fromPost:(ReaderPost *)post
+// Temporary duplication of [updateComment:withRemoteComment:] to address an issue in a frozen branch
+// We'll sort it out properly in develop.
+// - @aerych
+- (void)updateReaderComment:(Comment *)comment withRemoteComment:(RemoteComment *)remoteComment
```
I know this is a quick temp fix, but couldn't this just be:

```
[self updateComment:comment withRemoteComment:remoteComment];
comment.content = remoteComment.content;
```

aerych replied:
Ah it totally could have been. 🤦   Oh wellz.  Temp code ftw I guess.

ScoutHarris reviewed (approved):
Hey @aerych . All test cases check out. I just had one question/suggest for your consideration. Thanks much for fixing this!!

:shipit:

mokagio committed:
Update WordPressKit to version 4.37.1

To validate this fix locally, I run the tests for `CommentService` as
4.37.1 updates the Comment model.

mokagio reviewed (commented):
Thank you for the quick fix @aerych!

Do you think there's a unit test that we could have written that would have caught this issue?

When bumping WPKit to 4.37.1, I run the tests for `CommentService` locally before committing but I found the test coverage was sparse. Let me know if there's something I can do in that sense in the context of the "proper" fix in `develop`.

mokagio commented:
@aerych   this has been bundled as part of [17.8 beta 4 (17.8.0.4)](https://github.com/wordpress-mobile/WordPress-iOS/pull/16910).

Thanks for your work 🙌

aerych commented:
Thank you for wrangling the merge @mokagio!  🙇 

I'm not surprised by the sparse unit test coverage. Comments has rarely seen much attention in recent years.  Something we can look at improving as part of the ongoing comments unification work.
